sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/MapCameraManagerDelegate.kt

```diff
@@ -0,0 +1,286 @@
+   */
+  fun coordinateBoundsZoomForCameraUnwrapped(camera: CameraOptions): CoordinateBoundsZoom
+
+  /**
+   * Calculates a screen coordinate that corresponds to a geographical coordinate
+   * (i.e., longitude-latitude pair).
+   *
+   * The screen coordinate is in \link MapOptions#size platform pixels \endlink relative to the top left
+   * of the map (not of the whole screen).
+   *
+   * @param coordinate A geographical coordinate on the map to convert to a screen coordinate.
+   *
+   * @return Returns a screen coordinate on the screen in \link MapOptions#size platform pixels \endlink.
+   */
+  fun pixelForCoordinate(coordinate: Point): ScreenCoordinate
+
+  /**
+   * Calculates a geographical coordinate(i.e., longitude-latitude pair) that corresponds
+   * to a screen coordinate.
+   *
+   * The screen coordinate is in \link MapOptions#size platform pixels \endlink relative to the top left
+   * of the map (not of the whole screen).
+   *
+   * @param pixel A screen coordinate on the screen in \link MapOptions#size platform pixels \endlink.
+   *
+   * @return Returns a geographical coordinate corresponding to the ScreenCoordinate
+   * on the screen.
+   */
+  fun coordinateForPixel(pixel: ScreenCoordinate): Point
+
+  /**
+   * Calculates screen coordinates that corresponds to geographical coordinates
+   * (i.e., longitude-latitude pair).
+   *
+   * The screen coordinates are in \link MapOptions#size platform pixels \endlink relative to the top left
+   * of the map (not of the whole screen).
+   *
+   * @param coordinates A batch of geographical coordinates on the map to convert to screen coordinates.
+   *
+   * @return Returns a batch of screen coordinates on the screen in \link MapOptions#size platform pixels \endlink.
+   */
+  fun pixelsForCoordinates(coordinates: List<Point?>): List<ScreenCoordinate?>
+
+  /**
+   * Calculates geographical coordinates(i.e., longitude-latitude pair) that corresponds
+   * to screen coordinates.
+   *
+   * The screen coordinates are in \link MapOptions#size platform pixels \endlink relative to the top left
+   * of the map (not of the whole screen).
+   *
+   * @param pixel A batch of screen coordinates on the screen in \link MapOptions#size platform pixels \endlink.
+   *
+   * @return Returns a batch of geographical coordinates corresponding to the screen coordinates
+   * on the screen.
+   */
+  fun coordinatesForPixels(pixels: List<ScreenCoordinate?>): List<Point?>
+
+  /**
+   * Changes the map view by any combination of center, zoom, bearing, and pitch, without an animated transition.
+   * The map will retain its current values for any details not passed via the camera options argument.
+   *
+   * @param camera New camera options
+   */
+  fun setCamera(cameraOptions: CameraOptions)
+
+  /**
+   * Returns the current camera options
+   *
+   * @param padding Optional edge padding
+   */
+  fun getCameraOptions(padding: EdgeInsets?): CameraOptions
```

Comment:
can we make padding optional with providing a null as default parameter? I'm seeing a lot of changes related to this in this PR.

Reply:
Yes, we can.

Reply:
Please not that this parameter will be removed as part of getting `getCameraState()`.
My idea here - let's get this PR sooner to `main`, it will still be revisited later.